base/src/main/java/org/mozilla/jss/provider/javax/crypto/JSSTrustManager.java

```diff
@@ -219,17 +219,14 @@ public void checkCertChain(X509Certificate[] certChain, String keyUsage, Validit
 
         checkHostname(certChain, status);
 
-        if (!isTrustedPeer(certChain)) {
-            checkIssuerTrusted(certChain, status);
+        if (!isTrustedPeer(certChain) && isIssuerTrusted(certChain, status) && enableCertRevokeVerify) {        
+            certChainRevokeVerify(certChain, keyUsage, status);
```

Comment:
Generally I'd avoid creating a function with side effects (e.g. adding a reason into `status`) to make it easier to understand and reuse. This code probably could be written like this:
```
if (!isTrustedPeer(certChain)) {
    checkIssuerTrusted(certChain, status);
    if (!status.contains(UNKNOWN_ISSUER) && enableCertRevokeVerify) {
        certChainRevokeVerify(certChain, keyUsage, status);
    }
}
```
or this:
```
if (!isTrustedPeer(certChain)) {
    reason = checkIssuerTrusted(certChain);
    if (reason != null) {
        status.add(reason);
    } else if (enableCertRevokeVerify) {
        certChainRevokeVerify(certChain, keyUsage, status);
    }
}
```